hexrd/nf_config/experiment.py

```diff
@@ -33,33 +33,38 @@ def max_tth(self):
     def comp_thresh(self):
         key = 'experiment:comp_thresh'
         temp = self._cfg.get(key, None)
-        if temp is None:
-            return temp
-        elif np.logical_and(temp <= 1.0, temp > 0.0):
-            return temp
-        else:
-            raise RuntimeError('comp_thresh must be None or a number between 0 and 1')
-            
+        if temp is not None and np.any(
+            np.logical_or(temp > 1.0, temp <= 0.0)
+        ):
+            raise RuntimeError(
+                'comp_thresh must be None or a number between 0 and 1'
+            )
+
+        return temp
+
     @property
     def chi2_thresh(self):
         key = 'experiment:chi2_thresh'
         temp = self._cfg.get(key, None)
-        if temp is None:
-            return temp
-        elif np.logical_and(temp <= 1.0, temp > 0.0):
-            return temp
-        else:
-            raise RuntimeError('chi2_thresh must be None or a number between 0 and 1')
+        if temp is not None and np.any(
+            np.logical_or(temp > 1.0, temp <= 0.0)
+        ):
+            raise RuntimeError(
+                'chi2_thresh must be None or a number between 0 and 1'
+            )
+
+        return temp
 
 
     @property
     def misorientation(self):
         key = self._cfg.get('experiment:misorientation:use_misorientation')
         if key is True:
```

Comment:
`if key is True` is killing me...

Reply:
This should be typed to `bool` so that we can be sure `if key:` is safe to use instead. The danger is that it's a string, for example, and would evaluate to truthy. But we should just enforce that not to be possible.

Reply:
There's also no need to have the `key` variable here either.